src/components/Nodes/ClientConfig/index.js

```diff
@@ -131,23 +131,29 @@ class ClientConfig extends Component {
             <Tab label="Terminal" data-test-id="navbar-item-terminal" />
           </Tabs>
         </StyledAppBar>
+
         <TabContainer style={{ display: activeTab === 0 ? 'block' : 'none' }}>
           <VersionList
             client={client}
             handleReleaseSelect={handleReleaseSelect}
           />
         </TabContainer>
-        <TabContainer style={{ display: activeTab === 1 ? 'block' : 'none' }}>
-          <ErrorBoundary>
-            <DynamicConfigForm
-              clientName={client.name}
-              settings={getPluginSettingsConfig(client)}
-              handleClientConfigChanged={this.handleClientConfigChanged}
-              isClientRunning={isRunning}
-              clientConfigChanged={clientConfigChanged}
-            />
-          </ErrorBoundary>
-        </TabContainer>
+
+        {/* NOTE: MUI requires generating the ConfigForm from state each render */}
+        {activeTab === 1 && (
```

Comment:
nice hack, huh?

Reply:
Actually, this is good practice, conditionally rendering the component. The other two tabs may be considered 'hacks' for rendering upfront and showing/hiding the contents via css for performance.